x-pack/plugins/security_solution/public/management/cypress/e2e/endpoint/automated_response_actions.cy.ts

```diff
@@ -0,0 +1,105 @@
+import type { IndexedFleetEndpointPolicyResponse } from '../../../../../common/endpoint/data_loaders/index_fleet_endpoint_policy';
+import {
+  getAgentByHostName,
+  getEndpointIntegrationVersion,
+  reassignAgentPolicy,
+} from '../../tasks/fleet';
+
+describe.only('Automated Response Actions', () => {
+  const endpointHostname = Cypress.env(ENDPOINT_VM_NAME);
+  const unenrolledHost = 'Host unenrolled';
+  beforeEach(() => {
+    login();
+  });
+
+  describe('From alerts', () => {
+    let response: IndexedFleetEndpointPolicyResponse;
+    let initialAgentData: Agent;
+    let ruleId: string;
+    let ruleName: string;
+
+    before(() => {
+      getAgentByHostName(endpointHostname).then((agentData) => {
+        initialAgentData = agentData;
+      });
+
+      getEndpointIntegrationVersion().then((version) => {
+        createAgentPolicyTask(version, (data) => {
+          response = data;
+        });
+      });
+      loadRule(true).then((data) => {
+        ruleId = data.id;
+        ruleName = data.name;
+      });
+    });
+
+    after(() => {
+      if (initialAgentData?.policy_id) {
+        reassignAgentPolicy(initialAgentData.id, initialAgentData.policy_id);
+      }
+      if (response) {
+        cy.task('deleteIndexedFleetEndpointPolicies', response);
+      }
+      if (ruleId) {
+        cleanupRule(ruleId);
+      }
+    });
+
+    it('should have generated endpoint and rule', () => {
+      cy.visit(APP_ENDPOINTS_PATH);
+      cy.contains(endpointHostname).should('exist');
+
+      toggleRuleOffAndOn(ruleName);
+    });
+
+    it('should display endpoint automated response action in event details flyout ', () => {
+      visitRuleAlerts(ruleName);
+
+      closeAllToasts();
+      // wait for Endpoint to get isolated
+      cy.getByTestSubj('expand-event').first().click();
+      cy.getByTestSubj('endpointViewTab').click();
+      cy.getByTestSubj('endpoint-actions-notification').should('not.have.text', '0');
+      cy.getByTestSubj('endpoint-actions-results-table').within(() => {
+        cy.get('tbody tr').each(($tr, index) => {
+          cy.wrap($tr).within(() => {
+            cy.contains('Triggered by rule');
+            cy.getByTestSubj('endpoint-actions-results-table-column-hostname').then(($host) => {
+              // assert on the text
+              if ($host.text().includes(unenrolledHost)) {
+                cy.getByTestSubj('endpoint-actions-results-table-column-status').contains('Failed');
+                cy.getByTestSubj('endpoint-actions-results-table-expand-button').click();
+                cy.wrap($tr)
+                  .parent()
+                  .contains('The host does not have Elastic Defend integration installed');
+              } else {
+                cy.getByTestSubj('endpoint-actions-results-table-column-hostname').contains(
+                  endpointHostname
+                );
+                cy.getByTestSubj('endpoint-actions-results-table-column-status').contains(
```

Comment:
This one is not working, Instead of a timeout I would have to refresh the view to actually see the results. I will try to work it out in the evening.